ccxt.js

```diff
@@ -9354,20 +9360,30 @@ var btctradeua = {
     },
 
     parseTrade (trade, market) {
-        let timestamp = this.milliseconds (); // until we have a better solution for python
+        //let timestamp = this.milliseconds (); // until we have a better solution for python
+        let [d, month, y, , time] = trade['pub_date'].split (' ');
+        let [h, min, s] = time.split (':');
+        let months = ['января', 'февраля', 'марта', 'апреля', 'мая', 'июня', 'июля', 'августа', 'сентября', 'октября', 'ноября', 'декабря'];
+        let mon = months.indexOf (month);
+        let timestamp = Date.UTC (y, mon, d, h - 3, min, s);
         return {
             'id': trade['id'].toString (),
             'info': trade,
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
             'symbol': market['symbol'],
             'type': undefined,
-            'side': trade['type'],
+            'side': undefined,
             'price': parseFloat (trade['price']),
             'amount': parseFloat (trade['amnt_trade']),
         };
     },
 
+    parseTrades (trades, market = undefined) {
+        let dedupes = Object.values (trades).filter ( (val, idx, ar) => idx % 2 == 0);
+        return dedupes.map (trade => this.parseTrade (trade, market))
```

Comment:
Unfortunately, the mapping syntax isn't portable, therefore I will have to unfold this filter+map into basic c-style loops... It's the most difficult part to transpile, because functions don't work the same way in Python and PHP, as they do in JS. Those constructs are usually syntactically very different. So passing a callback is mostly sub-optimal in those languages (especially in PHP). This is why we don't use them anywhere in the derived code, even when we really want to (have to keep them as simple as possible → portable). But, again, don't worry, not a big deal at all, I'll merge it and we'll be fine.